pkg/component/plutono/plutono.go

```diff
@@ -84,6 +161,546 @@ func (p *plutono) WaitCleanup(ctx context.Context) error {
+								},
+							},
+							Resources: corev1.ResourceRequirements{
+								Requests: corev1.ResourceList{
+									corev1.ResourceCPU:    resource.MustParse("10m"),
+									corev1.ResourceMemory: resource.MustParse("32Mi"),
+								},
+								Limits: corev1.ResourceList{
+									corev1.ResourceMemory: resource.MustParse("400Mi"),
+								},
+							},
+						},
+					},
+					Volumes: []corev1.Volume{
+						{
+							Name: "plutono-dashboards",
+							VolumeSource: corev1.VolumeSource{
+								ConfigMap: &corev1.ConfigMapVolumeSource{
+									LocalObjectReference: corev1.LocalObjectReference{
+										Name: dashBoardConfigMapName,
+									},
+								},
+							},
+						},
+						{
+							Name: "plutono-datasources",
+							VolumeSource: corev1.VolumeSource{
+								ConfigMap: &corev1.ConfigMapVolumeSource{
+									LocalObjectReference: corev1.LocalObjectReference{
+										Name: dataSourceConfigMapName,
+									},
+								},
+							},
+						},
+						{
+							Name: "plutono-dashboard-providers",
+							VolumeSource: corev1.VolumeSource{
+								ConfigMap: &corev1.ConfigMapVolumeSource{
+									LocalObjectReference: corev1.LocalObjectReference{
+										Name: providerConfigMapName,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	if p.values.ClusterType == component.ClusterTypeSeed {
+		deployment.Labels = utils.MergeStringMaps(deployment.Labels, map[string]string{v1beta1constants.LabelRole: v1beta1constants.LabelMonitoring})
+		deployment.Spec.Template.Spec.AutomountServiceAccountToken = pointer.Bool(false)
+		deployment.Spec.Template.Spec.PriorityClassName = v1beta1constants.PriorityClassNameSeedSystem600
+		deployment.Spec.Template.Labels = utils.MergeStringMaps(deployment.Spec.Template.Labels, map[string]string{
+			v1beta1constants.LabelRole:                                         v1beta1constants.LabelMonitoring,
+			"networking.gardener.cloud/to-seed-prometheus":                     v1beta1constants.LabelNetworkPolicyAllowed,
+			gardenerutils.NetworkPolicyLabel("aggregate-prometheus-web", 9090): v1beta1constants.LabelNetworkPolicyAllowed,
+			gardenerutils.NetworkPolicyLabel("seed-prometheus-web", 9090):      v1beta1constants.LabelNetworkPolicyAllowed,
+		})
+		deployment.Spec.Template.Spec.Containers[0].Env = append(deployment.Spec.Template.Spec.Containers[0].Env, []corev1.EnvVar{
+			{Name: "PL_AUTH_BASIC_ENABLED", Value: "true"},
+			{Name: "PL_AUTH_DISABLE_LOGIN_FORM", Value: "false"},
+		}...)
+		deployment.Spec.Template.Spec.Containers[0].VolumeMounts = append(deployment.Spec.Template.Spec.Containers[0].VolumeMounts, corev1.VolumeMount{
+			Name:      "plutono-storage",
+			MountPath: "/var/plutono-storage",
+		})
+		deployment.Spec.Template.Spec.Volumes = append(deployment.Spec.Template.Spec.Volumes, corev1.Volume{
+			Name: "plutono-storage",
+			VolumeSource: corev1.VolumeSource{
+				EmptyDir: &corev1.EmptyDirVolumeSource{},
+			},
+		})
```

Comment:
This exists in both cases, the only difference is the size limit (to be clarified below/see other comment). Still, let's add the volume mount + volume above (first volume), and if needed only add the size limit field for shoots.

Reply:
Mount path is also different in the case of cluster type is shoot - 
```
		deployment.Spec.Template.Spec.Containers[0].VolumeMounts = append(deployment.Spec.Template.Spec.Containers[0].VolumeMounts, corev1.VolumeMount{
			Name:      "plutono-storage",
			MountPath: "/var/lib/plutono",
		})
```

Reply:
😵 Why is this @istvanballok @rickardsjp? I believe there is no explicit intention/reason? Can we streamline this?

Reply:
Thank you for catching this!

Actually this is a "bug", and having a streamlined Go component made it obvious. Unfortunately when we fixed it in

https://github.com/gardener/gardener/pull/7007/commits/42f781a28479828c170ba6e46fe9431034c021e5 of #7007,

we only fixed the occurrence for the shoot Grafana.

The original intention of the original authors for this emptyDir `grafana-storage` mount was to let Grafana write its files (before 7007 it was possible to temporarily create new dashboards with the admin role for Gardener operators) to an emptyDir mount, instead of the container's rw filesystem layer. It turns out that the emptyDir was mounted to an unused path and Grafana was happily using the container's file system. It "worked" and we did not notice that it worked in a not intended way. Unfortunately we only fixed one occurrence during 7007. Thank your for catching it and thank you for this cleanup!